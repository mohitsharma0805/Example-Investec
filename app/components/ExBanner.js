import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExBanner = ({ deviceType }) => {
  return (
    <View style={styles.contentView}>
      <Text style={styles.text}>{deviceType}</Text>
    </View>
  );
};

export default ExBanner;

const styles = StyleSheet.create({
  contentView: {
    justifyContent: 'center',
    height: 40,
    width: '100%',
    backgroundColor: config.colors.black,
  },
  text: { marginLeft: 20, color: config.colors.white, fontSize: 17 },
});
