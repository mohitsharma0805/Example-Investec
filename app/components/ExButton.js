import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExButton = ({ buttonStyle, text, textStyle, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, buttonStyle]}>
      <Text style={[styles.text, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ExButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    height: 50,
  },
  text: { color: config.colors.white, fontSize: 17 },
});
