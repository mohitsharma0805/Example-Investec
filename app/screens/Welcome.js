import React, { useState, useEffect } from 'react';
import { View, StyleSheet, SafeAreaView } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { StackActions } from '@react-navigation/native';
import { ExButton, ExTextInput, ExBanner } from '../components';
import config from '../config';
import { validateUserDetails } from '../controls/Validator';
import { selectBannerStatus } from '../state/Settings';
import { selectUser, saveUser } from '../state/User';

const Welcome = ({ navigation }) => {
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const bannerStatus = useSelector(selectBannerStatus);
  const user = useSelector(selectUser);

  useEffect(() => {
    detailsSavedSuccessfully();
  }, [user]);

  const detailsSavedSuccessfully = () => {
    if (Object.values(name).length > 0) {
      navigation.dispatch(StackActions.replace(config.routes.DASHBOARD));
    }
  };

  const nextClicked = () => {
    if (validateUserDetails(name)) {
      dispatch(saveUser({ name }));
    }
  };

  return (
    <SafeAreaView style={styles.contentView}>
      {bannerStatus && <ExBanner />}
      <View style={styles.container}>
        <ExTextInput
          placeholder={config.strings.ENTER_YOUR_NAME}
          autoCapitalize={config.strings.NONE}
          onChangeText={(text) => setName(text)}
        />
        <ExButton buttonStyle={styles.button} text={config.strings.NEXT} onPress={nextClicked} />
      </View>
    </SafeAreaView>
  );
};

export default Welcome;

const styles = StyleSheet.create({
  contentView: { flex: 1 },
  container: { marginTop: 150 },
  button: { marginTop: 50, backgroundColor: config.colors.appTheme, borderRadius: 8 },
});
