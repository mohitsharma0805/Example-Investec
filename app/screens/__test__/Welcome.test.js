import React from 'react';
import { shallow } from 'enzyme';
import { SafeAreaView } from 'react-native';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import { ExTextInput, ExButton, ExBanner } from '../../components';
import Welcome from '../Welcome';

describe('Welcome', () => {
  const dispatch = jest.fn();
  redux.useDispatch.mockReturnValue(dispatch);
  const navigation = {
    navigate: jest.fn(),
  };
  describe('Render UI with show banner', () => {
    const mockStore = configureStore();
    const store = mockStore({
      banner: true,
      user: {},
      deviceType: 'Device',
    });

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => store.getState().banner)
      .mockImplementationOnce(() => store.getState().user)
      .mockImplementationOnce(() => store.getState().deviceType);

    const wrapper = shallow(<Welcome navigation={navigation} />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(ExTextInput).length).toEqual(1);
      expect(wrapper.find(ExButton).length).toEqual(1);
      expect(wrapper.find(ExBanner).length).toEqual(1);
    });
    it('Simulate TextInput and Button', () => {
      const textInput = wrapper.find(ExTextInput).first();
      textInput.simulate('changeText', 'Mohit');

      const button = wrapper.find(ExButton);
      button.simulate('press');
      expect(button).toBeTruthy();
    });
  });
  describe('Render UI with banner hidden', () => {
    const mockStore = configureStore();
    const store = mockStore({
      banner: false,
      user: {},
    });

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => store.getState().banner)
      .mockImplementationOnce(() => store.getState().user);

    const wrapper = shallow(<Welcome navigation={navigation} />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(ExTextInput).length).toEqual(1);
      expect(wrapper.find(ExButton).length).toEqual(1);
      expect(wrapper.find(ExBanner).length).toEqual(0);
    });
    it('Simulate TextInput and Button', () => {
      const textInput = wrapper.find(ExTextInput).first();
      textInput.simulate('changeText', 'test');

      const button = wrapper.find(ExButton);
      button.simulate('press');
      expect(button).toBeTruthy();
    });
  });
});
