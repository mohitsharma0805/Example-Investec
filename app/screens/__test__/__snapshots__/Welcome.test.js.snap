// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Render UI with banner hidden renders the main view 1`] = `ShallowWrapper {}`;

exports[`Welcome Render UI with show banner renders the main view 1`] = `ShallowWrapper {}`;
